Extractor/Streamer.cs

```diff
@@ -335,7 +350,10 @@ public void HandleStreamedDatapoint(DataValue datapoint, VariableExtractionState
             }
 
             var buffDps = ToDataPoint(datapoint, node);
-            node.UpdateFromStream(buffDps);
+            if (StatusCode.IsGood(datapoint.StatusCode))
+            {
+                node.UpdateFromStream(buffDps);
+            }
```

Comment:
Similar situation here, we've already checked above, should we not remove this check?

Reply:
Same here, we only want to keep state for good points.